packages/cfd/src/Containers/cfd-password-modal/cfd-password-form/cfd-password-form.tsx

```diff
@@ -0,0 +1,288 @@
+                return onCancel();
+            }
+
+            return onForgotPassword();
+        };
+
+        const getAccountTitle = () => {
+            if (platform === CFD_PLATFORMS.DXTRADE) {
+                return getDxCompanies()[account_type.category as keyof TDxCompanies][
+                    account_type.type as keyof TDxCompanies['demo' | 'real']
+                ].short_title;
+            }
+
+            return account_title;
+        };
+
+        const handlePasswordInputChange = (
+            e: React.ChangeEvent<HTMLInputElement>,
+            handleChange: (el: React.ChangeEvent<HTMLInputElement>) => void,
+            validateForm: (values?: TCFDPasswordFormValues) => Promise<FormikErrors<TCFDPasswordFormValues>>,
+            setFieldTouched: (field: string, isTouched?: boolean, shouldValidate?: boolean) => void
+        ) => {
+            handleChange(e);
+            validateForm().then(() => {
+                setFieldTouched('password', true);
+            });
+        };
+
+        if (error_type === PASSWORD_ERRORS.RESET) {
+            return (
+                <React.Fragment>
+                    <div className='cfd-password-reset'>
+                        <div className='cfd-password-modal__content cfd-password-modal__content--password-reset'>
+                            <Text as='p' line_height='24' size='xs'>
+                                <Localize i18n_default_text='Please try again in a minute.' />
+                            </Text>
+                        </div>
+                        <Formik onSubmit={closeModal} initialValues={{}}>
+                            {({ handleSubmit }) => (
+                                <form onSubmit={handleSubmit}>
+                                    <FormSubmitButton
+                                        has_cancel={has_cancel_button}
+                                        cancel_label={cancel_button_label}
+                                        onCancel={handleCancel}
+                                        is_absolute={is_mobile}
+                                        label={button_label}
+                                    />
+                                </form>
+                            )}
+                        </Formik>
+                    </div>
+                </React.Fragment>
+            );
+        }
+
+        if (should_set_trading_password) {
+            return (
+                <PasswordStep
+                    platform={platform}
+                    error_message={error_message}
+                    validatePassword={validatePassword}
+                    submitPassword={submitPassword}
+                    has_mt5_account={has_mt5_account}
+                    is_real_financial_stp={is_real_financial_stp}
+                    handlePasswordInputChange={handlePasswordInputChange}
+                />
+            );
+        }
+
+        const showJurisdiction = () => {
```

Comment:
Does this need to be inside the component?

Reply:
Yes previously I've attempted to remove this however it ended up not working properly so I decided to just leave it there for the time being.